modules/ocf_kubernetes/templates/master/loadbalancer/haproxy.cfg.erb

```diff
@@ -20,31 +20,37 @@ defaults
 
 frontend lb-http
   bind 0.0.0.0:80
+  bind :::80
   mode http
 
+  # redirect *.ocf.io to *.ocf.berkeley.edu
+  http-request redirect prefix https://%[hdr(host),regsub(ocf\.io(:80)?$,ocf.berkeley.edu,i)] code 301 if { hdr(host) -m reg -i \.ocf\.io(:80)?$ }
+
   # redirect * to *.ocf.berkeley.edu
-  http-request redirect prefix https://%[hdr(host)].ocf.berkeley.edu unless { hdr_end(host) -i ocf.berkeley.edu ocf.io }
+  http-request redirect prefix https://%[hdr(host)].ocf.berkeley.edu code 301 unless { hdr(host) -m reg -i \.ocf\.berkeley\.edu(:80)?$ }
 
   # https redirect
-  redirect scheme https code 301 if !{ ssl_fc }
+  http-request redirect scheme https code 301 if !{ ssl_fc }
+
 
 frontend lb-https
   bind 0.0.0.0:443 ssl crt /etc/ssl/private/lb.ocf.berkeley.edu.pem
+  bind :::443 ssl crt /etc/ssl/private/lb.ocf.berkeley.edu.pem
   mode http
 
   # redirect *.ocf.io to *.ocf.berkeley.edu
-  http-request redirect prefix https://%[hdr(host),regsub(ocf.io$,ocf.berkeley.edu)] if { hdr_end(host) -i ocf.io }
+  http-request redirect prefix https://%[hdr(host),regsub(ocf\.io(:443)?$,ocf.berkeley.edu,i)] code 301 if { hdr(host) -m reg -i \.ocf\.io(:443)?$ }
 
   # redirect aliases, eg. sg.ocf.berkeley.edu -> sourcegraph.ocf.berkeley.edu
-  <%- @kubernetes_aliases.each do |service_name, aliases| -%>
-  http-request redirect prefix https://<%= service_name %> code 301 if { hdr(host) -i <%= aliases %> }
+  <%- @kubernetes_aliases.each do |service_name, alias| -%>
```

Comment:
based on the test failures it looks like we can't use `alias` as a variable name in erb for some reason

Reply:
yeah, `alias` is a protected keyword in ruby